torch/utils/_sympy/solve.py

```diff
@@ -117,8 +117,12 @@ def _try_isolate_lhs(
         # That is because we don't know whether we have mirror the operation or not.
         if not (isinstance(e, INEQUALITY_TYPES) and other.is_negative is None):
             # Divide both sides by 'other'.
-            lhs = lhs / other
-            rhs = rhs / other
+            if lhs.is_integer and rhs.is_integer and other.is_integer:
+                lhs = CleanDiv(lhs, other)
+                rhs = CleanDiv(rhs, other)
```

Comment:
Can you explain why this is OK to do for relational tests? This is only obviously sound when e is a sympy.Eq